src/gc.c

```diff
@@ -3137,12 +3137,12 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
             last_long_collect_interval = gc_num.interval;
         }
         else {
-            // reset interval to default, or at least half of live_bytes
+            // reset interval to half of live_bytes or at most max_collect_interval
             int64_t half = live_bytes/2;
             if (default_collect_interval < half && half <= max_collect_interval)
```

Comment:
Should we also add `if (half < default_collect_interval) gc_num.interval = default_collect_interval;`?